### DISCUSSION TITLE
Eclipse application with xtext grey screen

### DISCUSSION THREAD

**Discussion by Fnux8890** [eclipse-platform/eclipse.platform#1252] [Q&A]
Im trying to use Eclipse for a Model driven course at uni.
For this we are using xtext. Ive set up a base project to try and get stuff to work as a base before developing.
But when I run as a Eclipse application I get this ui.

<img width="500" alt="image" src="https://github.com/eclipse/xtext/assets/21344589/da9e2355-ea31-4815-bcd8-cb5ad9141932">

in my console I get this:

```console
WARNING: Using incubator modules: jdk.incubator.foreign, jdk.incubator.vector
SLF4J(W): No SLF4J providers were found.
SLF4J(W): Defaulting to no-operation (NOP) logger implementation
SLF4J(W): See https://www.slf4j.org/codes.html#noProviders for further details.
WARNING: Annotation classes from the 'javax.inject' or 'javax.annotation' package found.
It is recommended to migrate to the corresponding replacements in the jakarta namespace.
The Eclipse E4 Platform will remove support for those javax-annotations in a future release.
To suppress this warning, set the VM property: -Declipse.e4.inject.javax.warning=false
To disable processing of 'javax' annotations entirely, set the VM property: -Declipse.e4.inject.javax.disabled=true

!SESSION 2024-03-14 09:45:54.663 -----------------------------------------------
eclipse.buildId=4.30.0.20231201-1200
java.version=17.0.10
java.vendor=Homebrew
BootLoader constants: OS=macosx, ARCH=x86_64, WS=cocoa, NL=en_DK
Framework arguments:  -product org.eclipse.epp.package.dsl.product
Command-line arguments:  -product org.eclipse.epp.package.dsl.product -data /Users/bruger/Documents/Github/assignment2/../runtime-EclipseApplication -dev file:/Users/bruger/Documents/Github/assignment2/.metadata/.plugins/org.eclipse.pde.core/Eclipse Application/dev.properties -os macosx -ws cocoa -arch x86_64 -consoleLog

!ENTRY org.eclipse.jface 2 0 2024-03-14 09:46:00.871
!MESSAGE Keybinding conflicts occurred.  They may interfere with normal accelerator operation.
!SUBENTRY 1 org.eclipse.jface 2 0 2024-03-14 09:46:00.871
!MESSAGE A conflict occurred for COMMAND+SHIFT+T:
Binding(COMMAND+SHIFT+T,
	ParameterizedCommand(Command(org.eclipse.jdt.ui.navigate.open.type,Open Type,
		Open a type in a Java editor,
		Category(org.eclipse.ui.category.navigate,Navigate,null,true),
		WorkbenchHandlerServiceHandler("org.eclipse.jdt.ui.navigate.open.type"),
		,,true),null),
	org.eclipse.ui.defaultAcceleratorConfiguration,
	org.eclipse.ui.contexts.window,,,system)
Binding(COMMAND+SHIFT+T,
	ParameterizedCommand(Command(org.eclipse.lsp4e.symbolinworkspace,Go to Symbol in Workspace,
		,
		Category(org.eclipse.lsp4e.category,Language Servers,null,true),
		WorkbenchHandlerServiceHandler("org.eclipse.lsp4e.symbolinworkspace"),
		,,true),null),
	org.eclipse.ui.defaultAcceleratorConfiguration,
	org.eclipse.ui.contexts.window,,,system)
```

I've got open jdk 21 and 17 installed on my Mac and im trying to use the 17 one.
I have no idea of how to fix this or what's going on.

please help 🙏

**Comment by iloveeclipse** [comment]
Please try to update to https://download.eclipse.org/eclipse/downloads/drops4/I20240313-1800/.
There was a Mac painting issue due latest Mac OS changes.

**Comment by akurtakov** [comment]
This should be https://github.com/eclipse-platform/eclipse.platform.swt/issues/1012

**Comment by merks** [comment]
Note that this problem first cropped up in newer JREs where the java executable is built with a new version of the mac SDK.  So you either need to use the 4.32 version of SWT or you can use an older JRE/JDK for launcher.  I think the problem first cropped up in 17.0.10 while 17.0.9 still worked.

**Comment by Fnux8890** [comment]
how do I update to the [https://download.eclipse.org/eclipse/downloads/drops4/I20240313-1800/](https://download.eclipse.org/eclipse/downloads/drops4/I20240313-1800/) ?

I assume that a clean install of Eclipse would not work?
Im not very familiar with eclipse so I'm not really sure about how to upgrade the SDK or the SWT or other things regarding the Eclipse in general.

**Reply by merks** [reply]
I think it's actually hard (impossible) to update a downloaded product to the latest version of the Platform SDK's I-Build.  I see you are using the DSL product.   Normally one can provision a target platform that's independent from, i.e., newer than the installation itself, but I get the impression you are using the running installation as your target platform, so trying to learn how to do that quite a bit of work.

You're easiest option by far is to use an old JRE (17.0.9) until SimRel does its first M1 milestone (April 12th) and then update to that.  I.e., if you add this update site to your  Preferences ->  Install/Update -> Available Software Sites

https://download.eclipse.org/releases/milestone/

then you can update to each 2024-06 milestones, including the one with the fix for SWT.

You can choose a different JRE via Preferences -> Java -> Installed JREs

You can get the older 17.0.9 from from here:

https://adoptium.net/temurin/archive/?version=17

**Comment by Fnux8890** [comment]
I tried updating. (not sure to see how and if the update for the later build worked tho)

I'm still getting the grey screen

**Reply by merks** [reply]
Which version of SWT do you have installed now?  Help -> About... where 3.125 is the older one.

![image](https://github.com/eclipse-platform/eclipse.platform/assets/208716/cc835d9f-1902-4533-a715-e636cf124f96)

**Reply by Fnux8890** [reply]
ah. so Im still on the old one I see
<img width="746" alt="image" src="https://github.com/eclipse-platform/eclipse.platform/assets/21344589/2a8b90eb-d65b-478c-92c0-b067b2e750ad">

How do I use the [https://download.eclipse.org/eclipse/downloads/drops4/I20240313-1800/](https://download.eclipse.org/eclipse/downloads/drops4/I20240313-1800/) to update then. 
I assume that I didn't do I right then 🤔

**Reply by merks** [reply]
I did mention that it's **impossible** to update a packaged product that you got from download.eclipse.org or as installed with the installer to an Platform SDK integration build:

https://github.com/eclipse-platform/eclipse.platform/discussions/1252#discussioncomment-8783974

It's really not hard to download a JDK and configure the IDE to use it, as I described above.  That should take only a few minutes...

**Reply by Fnux8890** [reply]
oh. I missed your response for that. Sorry about that 😅
